File: cg/intro/core/circle.c
#include <math.h>
#include <GL/glut.h>

void draw() {
    int i, n = 50;
    float ang = 0;

    glClearColor(0, 0, 0, 0);
    glClear(GL_COLOR_BUFFER_BIT);
    glColor3f(1.0, 1.0 , 1.0);
    glBegin(GL_LINE_LOOP);
        for(i=0; i<n; i++){
            glVertex2f(0.5 * cos(ang), 0.5 * sin(ang));
            ang = ang+(2 * M_PI) / n;
        }
    glEnd();
    glFlush();
}

int main(int argc, char * argv []) {
    glutInit(&argc , argv);
    glutInitDisplayMode (GLUT_SINGLE | GLUT_RGB);
    glutInitWindowPosition (50, 100);
    glutInitWindowSize (400, 400);
    glutCreateWindow ("Circle");
    glutDisplayFunc (draw);
    glutMainLoop ();

    return 0;
}

File: cg/pyopengl/intro/circle.py
from OpenGL.GL import *
from OpenGL.GLUT import *
from math import pi, sin, cos

def draw():
    n = 50
    ang = 0
    ang_inc = (2 * pi) / n

    glClearColor(0, 0, 0, 0)
    glClear(GL_COLOR_BUFFER_BIT)
    glColor3f(1.0, 1.0, 1.0)
    
    glBegin(GL_LINE_LOOP)
    
    for i in range(n):
        x = 0.5 * cos(ang)
        y = 0.5 * sin(ang)
        glVertex2f(x, y)
        ang += ang_inc
    
    glEnd()
    glFlush()

def main():
    glutInit()
    glutInitDisplayMode(GLUT_SINGLE | GLUT_RGB)
    glutInitWindowPosition(50, 100)
    glutInitWindowSize(400, 400)
    glutCreateWindow("Circle")
    glutDisplayFunc(draw)
    glutMainLoop()

main()